state/upgrades_test.go

```diff
@@ -4995,6 +4995,219 @@ func (s *upgradesSuite) TestTranslateK8sServiceTypes(c *gc.C) {
+		"type": "kubernetes",
+		"auth-types": []interface{}{
+			"clientcertificate",
+			"oauth2",
+			"userpass",
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = cloudCredColl.Insert(bson.M{
+		"_id":       "localhost#admin#streetcred",
+		"owner":     "admin",
+		"cloud":     "localhost",
+		"name":      "streetcred",
+		"revoked":   false,
+		"auth-type": "oauth2withcert",
+		"attributes": map[string]string{
+			"Token": "this-is-my-token",
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	expectedClouds := []bson.M{{
+		"_id":  "localhost",
+		"name": "localhost",
+		"type": "kubernetes",
+		"auth-types": []interface{}{
+			"clientcertificate",
+			"oauth2",
+			"userpass",
+		},
+	}}
+
+	expectedCloudCreds := []bson.M{{
+		"_id":            "localhost#admin#streetcred",
+		"owner":          "admin",
+		"cloud":          "localhost",
+		"name":           "streetcred",
+		"revoked":        false,
+		"auth-type":      "oauth2",
+		"invalid":        false,
+		"invalid-reason": "",
+		"attributes": bson.M{
+			"Token": "this-is-my-token",
+		},
+	}}
+
+	f := func(pool *StatePool) error {
+		st := pool.SystemState()
+		return UpdateLegacyKubernetesCloudCredentials(st)
+	}
+	s.assertUpgradedData(c, f,
+		upgradedData(cloudColl, expectedClouds),
+		upgradedData(cloudCredColl, expectedCloudCreds),
+	)
+}
+
+func (s *upgradesSuite) TestUpdateLegacyKubernetesCloudCredentialsOAuth2Cert(c *gc.C) {
+	cloudColl, cloudCloser := s.state.db().GetRawCollection(cloudsC)
+	defer cloudCloser()
+	cloudCredColl, cloudCredCloser := s.state.db().GetRawCollection(cloudCredentialsC)
+	defer cloudCredCloser()
+
+	_, err := cloudColl.RemoveAll(nil)
+	c.Assert(err, jc.ErrorIsNil)
+	_, err = cloudCredColl.RemoveAll(nil)
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = cloudColl.Insert(bson.M{
+		"_id":  "localhost",
+		"name": "localhost",
+		"type": "kubernetes",
+		"auth-types": []interface{}{
+			"clientcertificate",
+			"oauth2",
+			"userpass",
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	err = cloudCredColl.Insert(bson.M{
+		"_id":       "localhost#admin#streetcred",
+		"owner":     "admin",
+		"cloud":     "localhost",
+		"name":      "streetcred",
+		"revoked":   false,
+		"auth-type": "oauth2withcert",
+		"attributes": map[string]string{
+			"ClientCertificateData": "data",
+			"ClientKeyData":         "data",
+		},
+	})
+	c.Assert(err, jc.ErrorIsNil)
+
+	expectedClouds := []bson.M{{
+		"_id":  "localhost",
+		"name": "localhost",
+		"type": "kubernetes",
+		"auth-types": []interface{}{
+			"clientcertificate",
+			"oauth2",
+			"userpass",
+		},
+	}}
+
+	expectedCloudCreds := []bson.M{{
+		"_id":            "localhost#admin#streetcred",
+		"owner":          "admin",
+		"cloud":          "localhost",
+		"name":           "streetcred",
+		"revoked":        false,
+		"auth-type":      "clientcertificate",
+		"invalid":        false,
+		"invalid-reason": "",
+		"attributes": bson.M{
+			"ClientCertificateData": "data",
+			"ClientKeyData":         "data",
```

Comment:
These expected attributes look to be the same as the ones created before the upgrade step is run.
Can we construct the "before" attributes to be what would be expected in the field so we can confirm the upgrade worked as expected. Also, don't use "data" for 2 different attributes.

Reply:
Done & Done